lib/io/file/delete.fz

```diff
@@ -25,59 +25,42 @@
 
 # delete -- effect wrapping the file delete operation
 #
-private delete(
-    dop Delete_Operation,
-    redef r effect_mode.val,
-    _ unit
-  ) : deletes, effect r is
+private delete0(dop Delete_Operation) : simpleEffect is
```

Comment:
probably better to use an additional arg `_ unit` since the name of effects are user visible.